# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/SharedSurfacesParent.cpp
# Commit: 11451cccc88c
# Full Hash: 11451cccc88c077e8fe763d4231bbf1af5226061
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-12 04:37:49
# Regressor Bug: 1869346
# File Overlap Count: 5
# Description:
#   Bug 1869346 - Rework how canvas accesses shared surfaces in the compositor process. r=gfx-reviewers,lsalzman
#   
#   This patch adds plumbing to allow sharing ExternalImageId via
#   SurfaceDescriptor. This will be used in a future patch by WebGL and
#   WebGPU to avoid extra copies. It also refactors how CanvasTranslator
# ==============================================================================

diff -r 8043d7693dba -r 11451cccc88c gfx/layers/ipc/SharedSurfacesParent.cpp
--- a/gfx/layers/ipc/SharedSurfacesParent.cpp	Mon Dec 11 18:15:34 2023 +0000
+++ b/gfx/layers/ipc/SharedSurfacesParent.cpp	Mon Dec 11 18:22:21 2023 +0000
@@ -206,6 +206,8 @@
     return;
   }
 
+  auto* renderThread = wr::RenderThread::Get();
+
   // Note that the destruction of a parent may not be cheap if it still has a
   // lot of surfaces still bound that require unmapping.
   for (auto i = sInstance->mSurfaces.Iter(); !i.Done(); i.Next()) {
@@ -217,8 +219,9 @@
     if (surface->HasCreatorRef() &&
         surface->RemoveConsumer(/* aForCreator */ true)) {
       RemoveTrackingLocked(surface, lock);
-      wr::RenderThread::Get()->UnregisterExternalImage(
-          wr::ToExternalImageId(i.Key()));
+      if (renderThread) {
+        renderThread->UnregisterExternalImage(wr::ToExternalImageId(i.Key()));
+      }
       i.Remove();
     }
   }